skglm/utils.py

```diff
@@ -457,3 +457,114 @@ def extrapolate(self, w, Xw):
         C = inv_UTU_ones / np.sum(inv_UTU_ones)
         # floating point errors may cause w and Xw to disagree
         return self.arr_w_[:, 1:] @ C, self.arr_Xw_[:, 1:] @ C, True
+
+
+@njit
+def _prox_vec(w, z, penalty, lipschitz):
+    # evaluate the vectorized proximal operator for the FISTA solver
+    # lipschitz stands for global lipschitz constant
+    n_features = w.shape[0]
+    for j in range(n_features):
+        w[j] = penalty.prox_1d(z[j], 1 / lipschitz, j)
+    return w
+
+
+@njit
+def spectral_norm2(X_data, X_indptr, X_indices, n_samples,
```

Comment:
the 2 is ambiguous here. 
rename the function to spectral_norm, return the sqrt of what's being computed, and square it when needed. It'll avoid surprises